### PR TITLE
Replace iobio for URL support

### PR DESCRIPTION
**Pull request by robertaboukhalil**
* Upgrade to samtools 1.17
* Generalize BamFile class so we can use the same structure for VCFs
* Replace iobio with biowasm URL support

To test:
* https://42basepairs.com/download/r2/genomics-data/alignments_HG002.bam
* chr20:60000

### CODE REVIEW

File: js/index_ribbon.js [hunk 1]
```diff
@@ -388,12 +388,6 @@ function open_any_url_files() {
   if (url_vars["perma"] != undefined) {
     read_permalink(url_vars["perma"]);
   }
-
```
Comment (robertaboukhalil) [inline js/index_ribbon.js:391]: Removing support for ?bam in the URL since we'll replace it with a session JSON url

File: js/index_ribbon.js [hunk 2]
```diff
@@ -5738,22 +5732,17 @@ function set_variant_info_text() {
...
+  if (!in_background) {
+    set_alignment_info_text();
+  }
 
   wait_then_run_when_bam_file_loaded();
-  _ribbon_settings.alignment_info_text = "Bam from url: " + url;
-  _ribbon_settings.bam_url = url;
-  set_alignment_info_text();
+  _Bam = new BamFile([url, url + ".bai"]);
+  await _Bam.mount();
```
Comment (robertaboukhalil) [inline js/index_ribbon.js:5744]: Combining these two very similar functions into 1

**Comment by MariaNattestad** [comment]
Weird, I got an error loading the same 42basepairs URL: https://42basepairs.com/download/r2/genomics-data/alignments_HG002.bam

**With coordinates: chr17:34000000**

Screenshot 2024-08-19 at 2 09 04 PM

**Comment by MariaNattestad** [comment]
Despite issue above, I'll just approve this so we can merge it so we can keep working and fix that one downstream.

**Review by MariaNattestad — APPROVED** [review]

**Pull request merged**